src/Shared/FileMatcher.cs

```diff
@@ -2440,6 +2441,12 @@ static string[] CreateArrayWithSingleItemIfNotExcluded(string filespecUnescaped,
             return files;
         }
 
+        private static bool IsSubdirectoryOf(string possibleChild, string possibleParent)
+        {
+            return possibleChild.StartsWith(possibleParent, StringComparison.OrdinalIgnoreCase) &&
+                   directorySeparatorCharacters.Contains(possibleChild[possibleParent.Length]);
```

Comment:
Hmm, this only works if the parent does not end with a directory separator. Maybe it's best to adjust the index based on that (worst case parent could end in multiple, different directory separators). Or maybe just assert it does not end in a separator. Even if the code calling this normalizes a certain way, I can easily see it drifting away.

Reply:
I think other things would go wrong if the normalization failed here, but it was easy enough to account for parent-ending-with-slash, so I did that.